File: backend/src/main/java/com/example/backend/services/AuthService.java
package com.example.backend.services;

import com.example.backend.dto.auth.ChangePasswordRequest;
import com.example.backend.dto.auth.ChangeUsernameRequest;
import com.example.backend.dto.auth.ForgotPasswordRequest;
import com.example.backend.dto.auth.LoginRequest;
import com.example.backend.dto.auth.LoginResponse;
import com.example.backend.dto.auth.RefreshTokenRequest;
import com.example.backend.dto.auth.RefreshTokenResponse;
import com.example.backend.dto.auth.RegisterRequest;
import com.example.backend.dto.auth.ResendOtpRequest;
import com.example.backend.dto.auth.ResetPasswordRequest;
import com.example.backend.dto.auth.VerifyChangePasswordRequest;
import com.example.backend.dto.auth.VerifyChangeUsernameRequest;
import com.example.backend.dto.auth.VerifyForgotPasswordOtpRequest;
import com.example.backend.dto.auth.VerifyLoginOtpRequest;
import com.example.backend.entities.User;

public interface AuthService {
	
//    User createUser(User user);

//    LoginResponse login(LoginRequest request);
	
	// Step 1: User submits registration details
    void registerRequest(RegisterRequest request);

    // Step 2: User verifies OTP and account is created
    User verifyRegistration(String email, String otp);

    // Login
    LoginResponse login(LoginRequest request);
    
    
    // Forgot Password
    void forgotPassword(ForgotPasswordRequest request);
    void verifyForgotPasswordOtp(
            VerifyForgotPasswordOtpRequest request
    );
    void resetPassword(ResetPasswordRequest request);
    
    
    // Change Password or Update Password
    void changePassword(ChangePasswordRequest request);

    void verifyChangePassword(VerifyChangePasswordRequest request);
    
    
    // Change Username or Update Username
    void changeUsername(ChangeUsernameRequest request);

    void verifyChangeUsername(VerifyChangeUsernameRequest request);
    
    
    
    // Login with OTP, JWTtoken generated only after verifying the OTP
    String loginRequest(LoginRequest request);

    LoginResponse verifyLoginOtp(VerifyLoginOtpRequest request);
    
    // Resend OTP for registration, forgot password, change password, change username, and login
    void resendOtp(ResendOtpRequest request);
    
    //Refresh Token
    RefreshTokenResponse refreshToken(
            RefreshTokenRequest request
    );
	



}

File: frontend/src/services/auth/authService.js
// import api from "../api";
import publicApi from "../publicApi";
import privateApi from "../privateApi";

import API_ENDPOINTS from "../../config/apiEndpoints";



export const registerUser = (data) =>
  publicApi.post(API_ENDPOINTS.REGISTER, data);

export const verifyRegistration = (data) =>
  publicApi.post(API_ENDPOINTS.VERIFY_REGISTRATION, data);

export const loginUser = (data) =>
  publicApi.post(API_ENDPOINTS.LOGIN, data);

export const verifyLogin = (data) =>
  publicApi.post(API_ENDPOINTS.VERIFY_LOGIN, data);

export const logoutUser = () =>
  privateApi.post(API_ENDPOINTS.LOGOUT);

export const forgotPassword = (data) =>
  publicApi.post(API_ENDPOINTS.FORGOT_PASSWORD, data);

export const verifyForgotPassword = (data) =>
  publicApi.post(API_ENDPOINTS.VERIFY_FORGOT_PASSWORD, data);

export const resetPassword = (data) =>
  publicApi.post(API_ENDPOINTS.RESET_PASSWORD, data);

export const changePassword = (data) =>
  privateApi.post("/auth/change-password/request", data);

export const verifyChangePassword = (data) =>
  privateApi.post("/auth/change-password/verify", data);

export const changeUsername = (data) =>
  privateApi.post("/auth/change-username/request", data);

export const verifyChangeUsername = (data) =>
    privateApi.post("/auth/change-username/verify", data);

// export const resetPassword = (data) =>
//   api.post("/auth/reset-password", data);

export const resendOTP = (data) =>
  publicApi.post("/auth/resend-otp", data);

export const refreshToken = (data) =>
  privateApi.post("/auth/refresh-token", data);

